filemonitor/FileMonitor.hpp

```diff
@@ -44,12 +47,16 @@ class FileMonitor final : public IFileMonitor {
                 } else {
                     std::cerr << "error!" << std::endl;
                 }
+
+                // unlock access to file
+                m_fileMutex.unlock();
             }
         }
         }
 
     private:
         fs::path m_filePath;
+        std::mutex m_fileMutex;
```

Comment:
why do you have a separate mutex for each object?

Reply:
mutexes reimplemented